tests/unit/test-cmd/test.build.js

```diff
@@ -37,6 +44,94 @@ describe('build', () => {
     );
   });
 
+  it('gives the correct name to an extension that is not localized', () => {
+    return withTempDir(
+      (tmpDir) =>
+        build({
+          sourceDir: fixturePath('minimal-web-ext'),
+          artifactsDir: tmpDir.path(),
+        })
+        .then((buildResult) => {
+          assert.match(buildResult.extensionPath,
+                       /minimal_extension-1\.0\.zip$/);
+          return buildResult.extensionPath;
+        })
+    );
+  });
+
+  it('gives the correct name to a localized extension', () => {
+    return withTempDir(
+      (tmpDir) =>
+        build({
+          sourceDir: fixturePath('minimal-localizable-web-ext'),
+          artifactsDir: tmpDir.path(),
+        })
+        .then((buildResult) => {
+          assert.match(buildResult.extensionPath,
+                       /name_of_the_extension-1\.0\.zip$/);
+          return buildResult.extensionPath;
+        })
+    );
+  });
+
+  it('checks locale file for malformed json', () => {
+    return withTempDir(
+      (tmpDir) => {
+        const messageFileName = path.join(tmpDir.path(), 'messages.json');
+        fs.writeFileSync(messageFileName,
+                '{"simulated:" "json syntax error"');
+        return getDefaultLocalizedName({
+          messageFile: messageFileName,
+          manifestData: manifestWithoutApps,
+        })
+        .then(makeSureItFails())
```

Comment:
This indentation is misleading because `.then()` is on the same level as the `return` block yet it's a continuation of the `return` statement. I suggest writing it like this:
````
return getDefaultLocalizedName(
  {
    messageFile: '...',
  })
  .then(makeSureItFails());
````
I know it's not pretty :) It will all be converted to async/await soon